OctopusDSC/DSCResources/cOctopusWorkerPool/cOctopusWorkerPool.psm1

```diff
@@ -0,0 +1,248 @@
+# Workers and Worker Pools appeared in 2018.7.0
+# Scoping to Spaces in 2019.1.0. TODO: version checking!
+
+# dot-source the helper file (cannot load as a module due to scope considerations)
+. (Join-Path -Path (Split-Path (Split-Path $PSScriptRoot -Parent) -Parent) -ChildPath 'OctopusDSCHelpers.ps1')
+
+function Get-TargetResource {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSDSCUseVerboseMessageInDSCResource", "")]
+    [OutputType([HashTable])]
+    param (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Url,
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateSet("Present", "Absent")]
+        [string]$Ensure,
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string]$WorkerPoolName,
+        [string]$WorkerPoolDescription,
+        [string]$SpaceId,
+        [PSCredential]$OctopusCredentials = [PSCredential]::Empty,
+        [PSCredential]$OctopusApiKey = [PSCredential]::Empty
+    )
+    $workerpool = Get-WorkerPool -Url $Url `
+        -WorkerPoolName $WorkerPoolName `
+        -OctopusCredentials $OctopusCredentials `
+        -OctopusApiKey $OctopusApiKey
+    $existingEnsure = 'Present'
+    if ($null -eq $workerpool) {
+        $existingEnsure = 'Absent'
+    }
+
+    $result = @{
+        Url                   = $Url;
+        Ensure                = $existingEnsure
+        WorkerPoolName        = $WorkerPoolName
+        WorkerPoolDescription = $WorkerPoolDescription
+        SpaceId               = $SpaceId
+        OctopusCredentials    = $OctopusCredentials
+        OctopusApiKey         = $OctopusApiKey
+    }
+
+    return $result
+}
+
+function Set-TargetResource {
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSDSCUseVerboseMessageInDSCResource", "")]
+    param (
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Url,
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateSet("Present", "Absent")]
+        [string]$Ensure,
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [string]$WorkerPoolName,
+        [string]$WorkerPoolDescription,
+        [string]$SpaceId,
+        [PSCredential]$OctopusCredentials = [PSCredential]::Empty,
+        [PSCredential]$OctopusApiKey = [PSCredential]::Empty
+    )
+
+    $currentResource = Get-TargetResource -Url $Url `
+        -Ensure $Ensure `
+        -WorkerPoolName $WorkerPoolName `
+        -OctopusCredentials $OctopusCredentials `
+        -OctopusApiKey $OctopusApiKey
+
+    if ($Ensure -eq "Absent" -and $currentResource.Ensure -eq "Present") {
```

Comment:
Drift in the `WorkerPoolDescription` field is not handled

Reply:
Indeed it isn't right now. Same for space membership.